pkg/activator/handler/handler.go

```diff
@@ -34,19 +33,36 @@ type ActivationHandler struct {
 	Logger    *zap.SugaredLogger
 	Transport http.RoundTripper
 	Reporter  activator.StatsReporter
+	Throttler *activator.Throttler
 }
 
 func (a *ActivationHandler) ServeHTTP(w http.ResponseWriter, r *http.Request) {
 	namespace := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderNamespace)
 	name := pkghttp.LastHeaderValue(r.Header, activator.RevisionHeaderName)
-
 	start := time.Now()
+	revID := activator.RevisionID{namespace, name}
+
+	// ActiveEndpoint() will block until the first endpoint is available.
+	ar := a.Activator.ActiveEndpoint(namespace, name)
+
+	err := a.Throttler.Try(revID, func() {
+		a.sendRequest(w, r, namespace, name, start, ar)
+	})
+	if err != nil {
+		if err == activator.OverloadMessage {
+			http.Error(w, activator.OverloadMessage.Error(), http.StatusServiceUnavailable)
+		} else {
+			w.WriteHeader(http.StatusInternalServerError)
+		}
+	}
+}
+
+func (a *ActivationHandler) sendRequest(w http.ResponseWriter, r *http.Request, namespace, name string, start time.Time, ar activator.ActivationResult) {
```

Comment:
Should this be called "proxyRequest"?

Can we get away with only passing w and r in here? We could easily handle the timing and the reporting in the outer function. WDYT?